src/result.alp

```diff
@@ -0,0 +1,57 @@
+{- Basic result type.
+ -}
+module result
+
+import_type option.option
+
+export_type result
+
+type result 'a 'b = Ok 'a | Err 'b
+
+let is_ok r =
+  match r with
+    Ok _ -> true
+  | Err _ -> false
+
+let is_ok r =
+  match r with
+    Ok _ -> false
+  | Err _ -> true
+
+test "Applying `is_ok` to Ok x should return true" =
+  assert.equal (true) (is_ok (Ok 1))
+
+let map f (Ok x) = Ok (f x)
+
+let map _ (Err x) = Err x
+
+test "mapping the identity function to Ok x should return Ok x" =
+  let f x = x in
+  assert.equal (Ok 1) (map f (Ok 1))
+
+let of_option opt e =
+  match opt with
+    Some x -> Ok x
+  | None -> Err e
```

Comment:
Could we can this from_option? That name is more common everywhere but OCaml and I think it is easier to discover as the name is closer to spoken English.